src/IR/LambdaLift.hs

```diff
@@ -182,24 +210,26 @@ liftLambdas (I.Prim p exprs t) = do
 liftLambdas lam@(I.Lambda _ _ t) = do
   let (vs, body) = I.collectLambda lam
       vs'        = zip vs $ fst (collectArrow t)
-  (liftedLamBody, lamFreeTypes) <-
-    descend $ newScope (catMaybes vs) >> liftLambdas body
+  lamName                       <- getFresh
+  fullName                      <- prependTrail lamName
+  (liftedLamBody, lamFreeTypes) <- descend (Just lamName) $ do
+    newScope (catMaybes vs)
+    liftLambdas body
   let liftedLam = I.makeLambdaChain
         (map (B.first Just) (M.toList lamFreeTypes) ++ vs')
```

Comment:
nit: `import Data.Bifunctor (first)` instead of using a qualified import, for consistency with other modules